Dashboard/app/js/lib/components/reusable/OfflineIndicator/index.jsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+
+export default class OfflineIndicator extends React.Component {
+  state = {
+    isOffline: false,
+  };
+
+  componentDidMount() {
+    this.timer = setInterval(() => {
+      fetch('https://www.google.com/generate_204', { mode: 'no-cors' })
+        .then(res => {
+          // with no-cors, res.status is 0 because the javascript ignores all response.
+          // we don't need the response, all we need to know is if the request went through or not
+          if (res.status !== 204) {
+            this.setState({ isOffline: false });
+          }
```

Comment:
https://stackoverflow.com/questions/58518098/fetch-request-to-httpswww-google-com-fails-obviously-with-cors#comment103363447_58518361

This explains this line